march_gait_selection/src/march_gait_selection/GaitSelection.py

```diff
@@ -1,296 +1,255 @@
+
+        if not os.path.isfile(subgait_path):
+            raise FileNotFoundError(file_path=subgait_path)
+
+        return subgait_path
+
+    def get_subgait_from_loaded_subgaits(self, gait_name, subgait_name):
+        """Get the sub gait data from gait and subgait name"""
+        return self.loaded_subgaits[gait_name][subgait_name]
+
     def set_gait_version_map(self, gait_version_map):
+        """Set gait version map and reload subgait files
+        :param gait_version_map:
+            list of gaits from yaml file
+        """
         self.gait_version_map = gait_version_map
         self.load_subgait_files()
 
     def set_subgait_version(self, gait_name, subgait_name, version):
-        if self.validate_version_name(gait_name, subgait_name, version):
+        """Set the sub gait version to a specific sub gait and reload subgait files
+        :param gait_name:
+            name of the gait in which the subgait is placed
+        :param subgait_name:
+            name of the subgait
+        :param version:
+            new version to set to parameter
+
+        :returns
+            if input was valid and version is set return True else False
+        """
+        if self.validate_version_file(self.gait_directory, gait_name, subgait_name, version):
             self.gait_version_map[gait_name][subgait_name] = version
             self.load_subgait_files()
             return True
         return False
 
+    def validate_gaits_in_map(self, gait_name=None, subgait_name=None):
+        """Validate the given name and subgait in the parsed map"""
+        if gait_name is not None:
+            if self.gait_version_map.get(gait_name) is None:
+                raise GaitNameNotFound(gait_name)
+
+            if subgait_name is not None:
+                if self.gait_version_map[gait_name].get(subgait_name) is None:
+                    raise SubgaitNameNotFound(subgait_name)
+
+        return True
+
+    @staticmethod
+    def validate_version_file(gait_directory, gait_name, subgait_name, version):
+        """Validate if the given gait name, subgait name and version form file path"""
+        if any(len(directory_name) == 0 for directory_name in (gait_name, subgait_name, version)):
+            return False
+
+        subgait_path = os.path.join(gait_directory, gait_name, subgait_name, version + '.subgait')
+        if os.path.isfile(subgait_path):
+            return True
+
+        return False
+
+    def validate_version_in_map(self, gait_map):
+        """Check if the given version in the gait mapping exists within the directory"""
+        for gait in gait_map:
+            for subgait in gait_map[gait]:
+                version = gait_map[gait][subgait]
+                if not self.validate_version_file(self.gait_directory, gait, subgait, version):
+                    return False
+        return True
+
+    def validate_gait_file(self, gait_name):
+        """Validate if the given gait name exists in the gait mapping
```

Comment:
I think this function checks whether the .gait file exists, not whether the gait exists in the mapping.
```suggestion
        """Validate that a valid .gait file exists for the given gait name
```